src/eos-metrics-instrumentation.c

```diff
@@ -78,6 +90,96 @@ static gint64 start_time;
 
 static EinsPersistentTally *persistent_tally;
 
+static gboolean
+get_os_version (char **name_out,
+                char **version_out)
+{
+    char *name = NULL;
+    char *version = NULL;
+    GError *error = NULL;
+    gboolean ret = FALSE;
+
+    GFile *osrelease_file = g_file_new_for_path (OSRELEASE_FILE);
+    GFileInputStream *filestream = g_file_read (osrelease_file, NULL, &error);
+    if (error)
+      goto out;
+
+    GDataInputStream *datastream = g_data_input_stream_new (G_INPUT_STREAM (filestream));
+
+    char *line;
+    while ((!name || !version) &&
+           (line = g_data_input_stream_read_line (datastream, NULL, NULL, &error)))
+      {
+        if (g_str_has_prefix (line, "NAME="))
+          name = line;
+        else if (g_str_has_prefix (line, "VERSION="))
+          version = line;
+        else
+          g_free (line);
+      }
+
+    if (error)
+      goto out;
+    if (!name || !version)
+      goto out;
+
+    /* According to os-release(5), these values can be quoted, escaped,
+     * etc. For simplicity, instead of doing the parsing on the client
+     * side, we do it on the server side. */
+    *name_out = g_strdup (name + strlen ("NAME="));
+    *version_out = g_strdup (version + strlen ("VERSION="));
+
+    ret = TRUE;
+
+ out:
+    g_free (name);
+    g_free (version);
+
+    if (error)
+      g_warning ("Error reading " OSRELEASE_FILE ": %s", error->message);
+
+    g_clear_error (&error);
+    g_clear_object (&datastream);
```

Comment:
We should be using the standard `AX_COMPILE_WARNINGS` macros to get it right out of the box.